submissions/ses3332002/memory-game/script.js

```diff
@@ -0,0 +1,140 @@
+      return;
+    };
+    doc.classList.remove(doc.classList[0]);
+    doc.classList.add(target.dataset.theme);
+  });
+  
+  function startGame() {
+    cardThemeSelected = document.querySelector('input[name="option"]:checked');
+    matchedCounter = 0;
+    prepareGameField();
+    switchControls();
+  };
+  
+  function switchControls() {
+    switchableControls.forEach((item) => {
+      item.disabled = !item.disabled;
+      item.classList.toggle("form_item-desabled");
+    });
+  };
+  
+  function prepareGameField() {
+    numbersArray = [];
+    for (let i=0; i < sizeSelector.value/2; i++) {
+      numbersArray[i] = i;
+    };
+    numbersArray = numbersArray.concat(numbersArray);
+
+    function shuffle(array) {
+      for (let i = array.length - 1; i > 0; i--) {
+        let j = Math.floor(Math.random() * (i + 1));
+        [array[i], array[j]] = [array[j], array[i]];
+      };
+    };
+    
+    shuffle(numbersArray);
+    
+    if (document.documentElement.clientHeight > document.documentElement.clientWidth) {
+      if (sizeSelector.value == 12) {
+        gameArea.classList.add("game_area-3columns");
+      } else {
+        gameArea.classList.add("game_area-4columns");
+      };
+    } else {
+      if (sizeSelector.value <= 16) {
+        gameArea.classList.add("game_area-4columns");
+      } else if (sizeSelector.value == 20){
+        gameArea.classList.add("game_area-5columns");
+      } else {
+        gameArea.classList.add("game_area-6columns");
+      };
+    };
+    let fragment = document.createDocumentFragment();
+    for (let i=0; i < sizeSelector.value; i++) {
+      let card = document.createElement("button");
+      card.classList.add("card");
+      card.innerHTML = `<div class="card__front" style="background-image: url(img/${cardThemeSelected.value}_bg.jpg);"></div><div class="card__back" style="background-image: url(img/${cardThemeSelected.value}_${numbersArray[i] + 1}.jpg);"></div>`;
+      fragment.append(card);
+      card.addEventListener("click", checkCard);
+    };
+    gameArea.append(fragment);
+    cardsArray = Array.from(document.querySelectorAll(".card"));
+  };
+  
+  function checkCard({target}) {
+    flipCard.call(target);
+    setTimeout(function() {
+      if (!cardSelected) {
+        cardSelected = cardsArray.indexOf(target) + 1;
+        lockCard.call(target);
+      } else if ((numbersArray[cardSelected - 1] == numbersArray[cardsArray.indexOf(target)])&&(cardSelected - 1 != cardsArray.indexOf(target))) {
+        vanishCard.call(cardsArray[cardSelected - 1]);
+        vanishCard.call(target);
+        cardSelected = "";
+        matchedCounter++;
+        if (matchedCounter == sizeSelector.value/2) {
+          hailWinner();
+        };
+      } else {
+        flipCard.call(cardsArray[cardSelected - 1]);
+        unlockCard.call(cardsArray[cardSelected - 1])
+        cardSelected = "";      
+        flipCard.call(target);
+      };
+    }, 400);
+    
+    function vanishCard() {
```

Comment:
It's better to move these functions (`vanishCard`, `flipCard`, `lockCard` and  `unlockCard`) outside the `checkCard` function because on the each call of the `checkCard` you create new functions. It's not ok for the performance.

Reply:
done